purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/common/Localization.kt

```diff
@@ -16,25 +16,57 @@ import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.descriptors.buildSerialDescriptor
 import kotlinx.serialization.encoding.Decoder
 import kotlinx.serialization.encoding.Encoder
+import java.util.Locale as JavaLocale
 
 /**
  * @property value The language tag of this locale, with an underscore separating the language from the region.
  */
 @InternalRevenueCatAPI
 @Serializable
 @JvmInline
-value class LocaleId(@get:JvmSynthetic val value: String) {
+value class LocaleId(@get:JvmSynthetic val value: String)
 
-    val language: String
-        get() = value.split('-', '_').getOrNull(0).orEmpty()
+@InternalRevenueCatAPI
+fun <V> Map<LocaleId, V>.getBestMatch(localeId: LocaleId): V? =
+    keys.getBestMatch(localeId)?.let { bestMatch -> get(bestMatch) }
 
-    val region: String
-        get() = value.split('-', '_').getOrNull(1).orEmpty()
-}
+/**
+ * Scripts inferred from the region.
+ */
+private val scriptByRegion = mapOf(
+    "CN" to "Hans",
+    "SG" to "Hans",
+    "MY" to "Hans",
+    "TW" to "Hant",
+    "HK" to "Hant",
+    "MO" to "Hant",
+)
 
+/**
+ * Returns the best match to [localeId] in this set, or null if no match is found.
+ */
 @InternalRevenueCatAPI
-fun LocaleId.languageOnly(): LocaleId =
-    LocaleId(language)
+fun Set<LocaleId>.getBestMatch(localeId: LocaleId): LocaleId? {
```

Comment:
I'm wondering if all this logic belongs here or in the UI module... But I think it might be ok to keep here for now...

Also, this is something we might also have to do for CustomerCenter... though in that case, it might be in the backend. cc @vegaro